### ISSUE TITLE
2016.11.2 - s3.put cannot create bucket

### ISSUE THREAD

**Issue by missionrulz** [saltstack/salt#39125]
### Description of Issue/Question

Using `2016.11.2` s3.put cannot create an empty bucket, it is always expecting a file to be provided.

From what I can see, this commit (https://github.com/saltstack/salt/commit/d9e5f2b4b490a10a63e670fe3f3dfd1219f2a867#diff-8165084beb8c6914b2331d79f2b4d9a2) fixed the problem, I'm not sure when this regression was introduced.

Somehow the `develop` branch seems to have the fix applied. https://github.com/saltstack/salt/blob/develop/salt/utils/s3.py#L152

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

To create an empty bucket as per the `2016.11.2` documentation.
`salt master s3.put testbucket`

Response:
```
root@saltmaster4:/home/sitehost# salt master s3.put krisdeleteme
[DEBUG   ] Configuration file path: /etc/salt/master
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Including configuration from '/etc/salt/master.d/salt-api.conf'
[DEBUG   ] Reading configuration from /etc/salt/master.d/salt-api.conf
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: master
[DEBUG   ] Missing configuration file: /root/.saltrc
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/master', 'master_master', 'tcp://127.0.0.1:4506', 'clear')
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] LazyLoaded local_cache.get_load
[DEBUG   ] Reading minion list from /var/cache/salt/master/jobs/01/cfa5cd351d41d415e59488ed5eb88489b98971aa2cee539c833bbfe12a43cb/.minions.p
[DEBUG   ] get_iter_returns for jid 20170201220301311752 sent to set(['master']) will timeout at 22:08:01.328706
[DEBUG   ] jid 20170201220301311752 return from master
[DEBUG   ] LazyLoaded nested.output
master:
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1412, in _thread_return
        return_data = executor.execute()
      File "/usr/lib/python2.7/dist-packages/salt/executors/direct_call.py", line 28, in execute
        return self.func(*self.args, **self.kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/modules/s3.py", line 260, in put
        role_arn=role_arn)
      File "/usr/lib/python2.7/dist-packages/salt/utils/s3.py", line 143, in query
        response = result.content
    UnboundLocalError: local variable 'result' referenced before assignment
[DEBUG   ] jid 20170201220301311752 found all minions set(['master'])
```

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)

```
Salt Version:
           Salt: 2016.11.2
 
Dependency Versions:
           cffi: 1.8.3
       cherrypy: unknown
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: 0.24.0
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: 0.24.1
         Python: 2.7.12 (default, Jul  1 2016, 15:12:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-45-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by rallytime** [comment]
Hi @missionrulz - thanks for filing this issue. It looks to me like we just need to back-port the fix from #36714 from the develop branch to the 2016.11 branch. I can do that today.

**Comment by rallytime** [comment]
@missionrulz Please give the change in #39133 a try. That should fix this up for you.

**Comment by missionrulz** [comment]
Hi @rallytime, I can confirm that applying the back-port has fixed this and I am now able to create an empty s3 bucket.

Thanks for the quick response.

**Comment by rallytime** [comment]
@missionrulz Great! Thanks for testing that so quickly. I just merged that back-porting pull request into 2016.11. It will be available in the 2016.11.3 release. I'll go ahead and close this up.
